acme/acme/client.py

```diff
@@ -612,7 +614,9 @@ def revoke(self, cert: jose.ComparableX509, rsn: int) -> None:
         :raises .ClientError: If revocation is unsuccessful.
 
         """
-        self._revoke(cert, rsn, self.directory[messages.Revocation])
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore', '.* non-string keys', DeprecationWarning)
+            self._revoke(cert, rsn, self.directory[messages.Revocation])
```

Comment:
We'll have to do something like this long term right? If so, I think we should just do it now.
```suggestion
        self._revoke(cert, rsn, self.directory['revoke-cert'])
```